Refactoring of introrob. Replaced IplImage to cv::Mat

fqez opened the pull request:
Replacement of the data type of the images handled by introrob. Removed the old type IplImage to the new and more efficient type cv::Mat

abhi-kumar commented:
@fqez in this commit we will also be needing to change the files at "src/stable/libs" folder as those are based on OpenCV1's IplImage type, unless we change that this wont be that helpful. Alternatively we can include opencv2 headers as mentioned in the pull request #40 . Because after making the changes made by this pull request I am getting build error stating, "Mat had not been declared" , "Mat is not a member of cv".

fqez commented:
Hi @abhi-kumar I'm not sure how to refactor the files inside src/stable/libs/visionlib, it's not a simple change.. but yes, it should be modified. About the includes, I've tried to compile the newest version of the code with my commit and it works fine :confused:. However, I've changed the includes of the API.h file with the following:

```
replaced:
#include <cv.h>
#include <highgui.h>
for:
#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
```

and it compiles well also, so I will commit this changes to this pull request so you can test it and tell me if it causes you problems.
Thanks for the answer!

abhi-kumar commented:
@fqez You got my comment wrong with respect to the not-compiling thing. What I mean to say is that at later stages while linking, some of the components use vision libs(colospaces, etc), may not be true in case of introrob. So for that linkage to be proper we need to change the vision libs. Once your commit is accepted I will fork it and change the visionlibs after proper discussion with you and rest of the team. As soon as yours is accepted I will test and make further changes and mention it here :)
